test/utils/node/commons.ts

```diff
@@ -0,0 +1,265 @@
+		});
+
+		it('should return if the directory exists', () => {
+			pathExistsSyncStub.returns(true);
+			return expect(createDirectory(defaultInstallationPath)).to.not.throw;
+		});
+
+		it('should create directory if it does not exists', () => {
+			pathExistsSyncStub.returns(false);
+			ensureDirSync.returns();
+			return expect(createDirectory(defaultInstallationPath)).to.not.throw;
+		});
+	});
+
+	describe('#validURL', () => {
+		it('should throw error if url is invalid', () => {
+			return expect(() => validURL('dummy://download.lisk.io')).to.throw(
+				'Invalid URL: dummy://download.lisk.io',
+			);
+		});
+
+		it('should not throw error if url is valid', () => {
+			return expect(validURL(`https://downloads.lisk.io/lisk/`)).to.not.throw;
+		});
+	});
+
+	describe('#getVersionToUpgrade', () => {
+		beforeEach(() => {
+			sandbox.stub(release, 'getLatestVersion').resolves('2.0.0');
+		});
+
+		it('should return version if specified', async () => {
+			const version = '2.0.0';
+			const result = await getVersionToUpgrade(NETWORK.MAINNET, version);
+			return expect(result).to.equal(version);
+		});
+
+		it('should return latest version if version is not specified', async () => {
+			const result = await getVersionToUpgrade(NETWORK.MAINNET);
+			return expect(result).to.equal('2.0.0');
+		});
+	});
+
+	describe('#backupLisk', () => {
+		let execStub: any = null;
+		beforeEach(() => {
+			sandbox.stub(fsExtra, 'emptyDirSync').returns(null);
+			execStub = sandbox.stub(workerProcess, 'exec');
+		});
+
+		it('should backup the lisk installation', async () => {
+			execStub.resolves({ stdout: '', stderr: null });
+			return expect(backupLisk(defaultInstallationPath)).to.not.throw;
+		});
+
+		it('should throw error of failed to backup', async () => {
+			execStub.resolves({ stdout: null, stderr: 'failed to move' });
+			return expect(backupLisk(defaultInstallationPath)).rejectedWith(
+				'failed to move',
+			);
+		});
+	});
+
+	describe('#upgradeLisk', () => {
+		let execStub: any = null;
+		beforeEach(() => {
+			sandbox.stub(fsExtra, 'mkdirSync').returns(null);
+			sandbox.stub(fsExtra, 'emptyDirSync').returns(null);
+			execStub = sandbox.stub(workerProcess, 'exec');
+		});
+
+		it('should throw error if failed to upgrade', async () => {
+			execStub.resolves({ stdout: '', stderr: 'failed to copy' });
+
+			return expect(
+				upgradeLisk(defaultInstallationPath, 'test', NETWORK.MAINNET, '1.0.0'),
+			).to.rejectedWith('failed to copy');
+		});
+
+		it('should throw error of failed to backup', async () => {
+			execStub.resolves({ stdout: '', stderr: null });
+			return expect(
+				upgradeLisk(defaultInstallationPath, 'test', NETWORK.MAINNET, '1.0.0'),
+			).to.not.throw;
+		});
+	});
+
+	describe('#validateVersion', () => {
+		let releaseStub: any = null;
+		beforeEach(() => {
+			releaseStub = sandbox.stub(release, 'getLatestVersion');
+		});
+
+		it('should throw if version is invalid', async () => {
```

Comment:
It should either consistently use `async` or not

Reply:
depends, if the function is async or not rite?

Reply:
in this below, you are returning, so it's a Promise.
If we gonna use `async` keyword, it doens't need return.
For now, i think it's good not to use `async` keyword